tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -74,15 +77,39 @@
 import py4j.GatewayServer;
 
 import static com.linkedin.tony.Constants.*;
-import static com.linkedin.tony.TonyConfigurationKeys.MLFramework;
+import static com.linkedin.tony.TonyConfigurationKeys.*;
+import static com.linkedin.tony.Utils.*;
 
 
 public class TonyApplicationMaster {
   private static final Log LOG = LogFactory.getLog(TonyApplicationMaster.class);
 
+  /**
+   * Metadata of jobs
+   */
   private ApplicationAttemptId appAttemptID = null;
-  private String appIdString;
+  private static String appIdString;
   private String amHostPort;
+  private static FileSystem fs;
+  private static String tonyHistoryFolder;
+  private static Path jobDir;
+
+  public static class Metadata {
```

Comment:
any reason to put this as a nested class instead of a new class?

Reply:
I was inspired by ask-hadoop-fetcher project. My reasons were that it's too trivial to make a separate class, and only TonyAM was interacting with this class at that point. But now since https://github.com/linkedin/TonY/pull/79/files/d28348e16791732b06f424d33f8a0ff8638ba51b#r233199638, we can refactor this to a separate class.

Reply:
`ask-hadoop-fetcher ` was a purely hackday project.. not a good reference..

Reply:
Yep, fixed: https://github.com/linkedin/TonY/pull/79/commits/6ba7908b4d3312f24a810d97f8a82fa46e6c6a8e.